exonum/src/blockchain/block.rs

```diff
@@ -19,8 +19,50 @@ use crate::{
     crypto::Hash,
     helpers::{Height, ValidatorId},
     messages::{Precommit, Verified},
-    proto,
+    proto::{self, BinaryMap},
 };
+use exonum_merkledb::BinaryValue;
+use std::borrow::Cow;
+
+/// Trait that represents key in block header entry map. Provide
+/// mapping between `NAME` of the entry and its value.
+///
+/// # Usage
+///
+/// see [`Block::get_entry()`].
+///
+/// [`Block::get_entry()`]: struct.Block.html#method.get_entry
+pub trait BlockHeaderKey {
+    /// Key name.
+    const NAME: &'static str;
+    /// Type of the value associated with this key.
+    type Value: BinaryValue;
+}
+
+impl BlockHeaderKey for ValidatorId {
```

Comment:
Not sure about this. We store not any abstract validator ID, but the proposer ID. I think it's better to create a separate structure for it.